### PR TITLE
Update: Persist items using IndexedDB

### LINKED ISSUE #3: Implement basic item storage

### PR DESCRIPTION
**Pull request by linuxwolf**
Fixes #3

### CODE REVIEW

**Comment by linuxwolf** [comment]
As the title indicates, this is still a work in progress.  Namely:

* Need a couple of airgap tests that verify things really are saved 
* Refactor how the keystore is saved/loaded
* Use transactions where multiple tables are impacted

**Comment by linuxwolf** [comment]
@jimporter @devinreams @m8ttyB 

This is good enough for review.  There's more tests I'd like, but I don't think they're critical to get before merging to master.

**Comment by linuxwolf** [comment]
@jimporter @m8ttyB ok, *now* you can review.

The latest change makes it at least **possible** to operate in node.js, so we don't have to get karma tests on `lockbox-extension` before merging this.

File: lib/localdatabase.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
+/*!
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+const Dexie = require("dexie"),
+      indexedDB = require("fake-indexeddb"),
+      IDBKeyRange = require("fake-indexeddb/lib/FDBKeyRange");
```
Comment (jimporter) [inline lib/localdatabase.js]: These are just for tests, right? Or would you need them if you were running this via node?

I wonder if there's a better way to inject fake IndexedDB APIs, but this is probably fine...

Comment (linuxwolf) [reply lib/localdatabase.js]: It's needed to run in node, regardless of testing.

There's not a better way that I've found, and this kind of pattern is pretty common in modules that have to do some juggling between node and web.

**Comment by linuxwolf** [comment]
Merging without more review -- just a rebase and squash

**Pull request merged**